web/src/app/api/[...path]/route.ts

```diff
@@ -0,0 +1,124 @@
+import { NextRequest, NextResponse } from "next/server";
+
+/* NextJS is annoying and makes use use a separate function for 
+each request type >:( */
+
+export async function GET(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function POST(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function PUT(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function PATCH(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function DELETE(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function HEAD(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function OPTIONS(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+async function handleRequest(request: NextRequest, path: string[]) {
+  if (process.env.NODE_ENV !== "development") {
+    return NextResponse.json(
+      {
+        message:
+          "This API is only available in development mode. In production, something else (e.g. nginx) should handle this.",
+      },
+      { status: 404 }
+    );
+  }
+
+  try {
+    const backendUrl = new URL(`http://localhost:8080/${path.join("/")}`);
+    const requestBody = request.body
+      ? JSON.stringify(await request.json())
```

Comment:
Believe we need to add additional handling for different content types (specifically I know we're using formData, url params in a couple places)

Reply:
Added handling for formData! url params should be handled above 👀